llmfoundry/utils/builders.py

```diff
@@ -440,7 +440,9 @@ def build_tokenizer(
             int(1e30),
         )
 
-    if not hasattr(tokenizer, 'eos_token') or tokenizer.eos_token is None:
+    if not hasattr(
+            tokenizer, 'eos_token'
+    ) or tokenizer.eos_token is None:  # type: ignore (sometime's it's not none but that's ok too)
```

Comment:
Why does this give a pyright error? Is it because `tokenizer.eos_token` is type `str` instead of `str | None`.

Reply:
I think so, I've seen this in a couple other places and it always claims it's never None but sometimes is.